docs/design/proposals/healer-controller.md

```diff
@@ -0,0 +1,120 @@
+# Introduction
+There is [external health monitoring](https://github.com/kubernetes-csi/external-health-monitor/) which does mostly similar stuff, but actions are not in the scope of it, all it can do is log the health condition of the mounts.
+
+# Volume Healer Controller
+Before we begin it is noteworthy to know that the existing healer logic remains the same.
+
+**Now the obvious question we expect is "what do we achieve with the new controller?"**
+* Moving the k8s specific code out of ceph-csi, to make ceph-csi orchestrator independent.
+* Achieve a way to monitor the health of NBD volumes.
+* We currently do not have a way to act on failed/crashed NBD volumes with the existing healer logic. Once the controller detects that the volume is not healthy, it will have an opportunity to act on it.
+* This will be a generic framework for any userspace mounters added in the future, example ceph-fuse
+
+# Details
+
+                    .-------------------.
+                    |     CSI-Addons    |
+                    | Healer Controller |
+                    '-------------------'
+                              |
+                              | gRPC
+                              |
+            .---------+------------------------------.
+            |         |                              |
+            |  .------------.        .------------.  |
+            |  | CSI-Addons |  gRPC  |    CSI     |  |
+            |  |  side-car  |--------| NodePlugin |  |
+            |  '------------'        '------------'  |
+            | CSI-driver Pod                         |
+            '----------------------------------------'
+
+## CSI-Addons Healer Controller
+* The healer controller will be implemented as part of the CSI-Addons controller driver. It is the one who triggers controller RPC to check the health condition of the CSI volumes.
+* Watch for the `CSIAddonsNode` object, if the object is created/modified by the side-car do below operations
+    * Fetch all the volume attachments (VA) list
+    * Filter the volume attachments list through matching driver name and status attached
+    * For each volume attachment get the respective PV information and check the criteria of PV Bound (also we can have configuration options like mounter type)
+    * Send the RPC request for side-car to check if the volume is healthy along with the PV details details
+    * Log the final response returned from the side-car
+* The healer controller periodically triggers the health status checks of the CSI volumes via the gRPC calls.
```

Comment:
I do not think a regular check is needed, you refer to https://github.com/kubernetes-csi/external-health-monitor/ for that in the top of the document